src/js/HScrollMenu.js

```diff
@@ -1,19 +1,36 @@
 import React from 'react';
+import store from './store'
+import { setPendingAppLaunch } from './actions'
 import HScrollMenuItem from './HScrollMenuItem'
 
 export default class HScrollMenu extends React.Component {
     constructor(props) {
         super(props);
+        this.onSelection = this.onSelection.bind(this);
+    }
+
+    onSelection(appID, cmdID, menuID) {
+        if (this.props.pendingAppLaunch) {
+            return;
+        }
+
+        this.props.onSelection(appID, cmdID, menuID);
+        store.dispatch(setPendingAppLaunch(appID));
     }
 
     render() {
         let menuItems = this.props.data.map((menuItem) => {
+            if (menuItem.appID === this.props.pendingAppLaunch) {
```

Comment:
Seems that this sort of logic should be in the `AppList` file, rather than the base menu class.